jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -616,6 +618,46 @@ public boolean removeEventListener(EventListener listener)
         return false;
     }
 
+    @Override
+    public void initializeDefaults(Attributes attributes)
+    {
+        for (String keyName : attributes.getAttributeNameSet())
+        {
+            Object value = attributes.getAttribute(keyName);
+            if (LOG.isDebugEnabled())
+                LOG.debug("init {}: {}", keyName, value);
+
+            switch (keyName)
+            {
+                case Deployable.TEMP_DIR -> setTempDirectory(IO.asFile(value));
+                case Deployable.CONTEXT_PATH -> setContextPath((String)value);
+                case Deployable.DEFAULT_CONTEXT_PATH -> setDefaultContextPath((String)value);
+                default -> initializeDefault(keyName, value);
+            }
+        }
+        initializeDefaultComplete();
+    }
+
+    /**
+     * Called for each attribute key encountered during the
+     * {@link Deployable#initializeDefaults(Attributes)} processing.
+     *
+     * @param keyName the key name
+     * @param value the value
+     */
+    protected void initializeDefault(String keyName, Object value)
+    {
+    }
+
+    /**
+     * Called after all attributes are processed via
+     * {@link Deployable#initializeDefaults(Attributes)}, to allow
+     * any kind of extra processing of the configuration.
+     */
+    protected void initializeDefaultComplete()
```

Comment:
Should be renamed to `initializeDefaultsComplete()` (note the `s` for the plural), since the `Deployable` method has the plural.

Reply:
Done.